docker/docker_image_dest.go

```diff
@@ -361,6 +363,13 @@ func (d *dockerImageDestination) TryReusingBlob(ctx context.Context, info types.
 
 		// Checking candidateRepo, and mounting from it, requires an
 		// expanded token scope.
+
+		if crossRegistry {
+			// found following blob in different registry but we need to check blob presence against the registry
+			// where we are planning to push, hence switch back the candidate repo to the one where we are planning to push
+			candidateRepo, _ = parseBICLocationReference(types.BICLocationReference{Opaque: string(d.ref.ref.Name())})
```

Comment:
(This could just be `candidateRepo = d.ref.ref.Name()` — and it should have happened instead of the `crossRegistry = true` assignment already, so that the “Already tried the primary destination” check applies to the cross-registry situation as well.)

Reply:
ackd.